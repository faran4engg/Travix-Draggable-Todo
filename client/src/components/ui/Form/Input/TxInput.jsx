import React from "react";
import { Form } from "semantic-ui-react";
import PropTypes from "prop-types";

const TxInput = ({ field, form: { touched, errors }, ...props }) => {
  return (
    <React.Fragment>
      <Form.Input
        {...field}
        {...props}
        error={touched[field.name] && errors[field.name]}
      />
    </React.Fragment>
  );
};

TxInput.propTypes = {
  field: PropTypes.object,
  form: PropTypes.object
};
export default TxInput;
