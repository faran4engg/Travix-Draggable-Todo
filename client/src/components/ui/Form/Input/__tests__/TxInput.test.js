import React from "react";
import { Form } from "semantic-ui-react";
import { Formik, Field } from "formik";
import { shallow } from "enzyme";
import TxInput from "../TxInput";

describe("TxInput", () => {
  it("should render correctly ", () => {
    const component = shallow(
      <Formik>
        {() => (
          <Form >
            <Field
              label="Title"
              type="text"
              placeholder="Title"
              name="title"
              value="Sample Title"
              component={TxInput}
            />
          </Form>
        )}
      </Formik>
    );

    expect(component).toMatchSnapshot();
  });
});
