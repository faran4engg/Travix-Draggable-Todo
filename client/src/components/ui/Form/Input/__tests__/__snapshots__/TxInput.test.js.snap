// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TxInput should render correctly  1`] = `ShallowWrapper {}`;
